src/JBCCIPSucker.sol

```diff
@@ -0,0 +1,196 @@
+        }
+
+        // approve the Router to spend tokens on contract's behalf. It will spend the amount of the given token
+        IERC20(token).approve(address(router), amount);
+
+        // TODO: Handle this messageId, maybe necessary
+        // Send the message through the router and store the returned message ID
+        /* messageId =  */
+        router.ccipSend{value: fees}({destinationChainSelector: remoteChainSelector, message: evm2AnyMessage});
+
+        // Refund remaining balance.
+        (bool sent,) = msg.sender.call{value: msg.value - fees}("");
+        if (!sent) revert FailedToRefundFee();
+
+        // Emit an event for the relayers to watch for.
+        emit RootToRemote(_root, token, _index, nonce);
+    }
+
+    /// @notice Construct a CCIP message.
+    /// @dev This function will create an EVM2AnyMessage struct with all the necessary information for programmable tokens transfer.
+    /// @param _receiver The address of the receiver.
+    /// @param _root The root to be sent.
+    /// @param _token The token to be transferred.
+    /// @param _amount The amount of the token to be transferred.
+    /// @param _feeTokenAddress The address of the token used for fees. Set address(0) for native gas.
+    /// @return Client.EVM2AnyMessage Returns an EVM2AnyMessage struct which contains information for sending a CCIP message.
+    function _buildCCIPMessage(
+        address _receiver,
+        JBMessageRoot memory _root,
+        address _token,
+        uint256 _amount,
+        address _feeTokenAddress
+    ) private pure returns (Client.EVM2AnyMessage memory) {
+        // Set the token amounts
+        Client.EVMTokenAmount[] memory tokenAmounts = new Client.EVMTokenAmount[](1);
+        tokenAmounts[0] = Client.EVMTokenAmount({token: _token, amount: _amount});
+        // Create an EVM2AnyMessage struct in memory with necessary information for sending a cross-chain message
+        return Client.EVM2AnyMessage({
+            receiver: abi.encode(_receiver), // ABI-encoded receiver address
+            data: abi.encode(_root), // ABI-encoded string
+            tokenAmounts: tokenAmounts, // The amount and type of token being transferred
+            extraArgs: Client._argsToBytes(
+                // Additional arguments, setting gas limit
+                Client.EVMExtraArgsV1({gasLimit: 300_000})
+            ),
+            // Set the feeToken to a feeTokenAddress, indicating specific asset will be used for fees
+            feeToken: _feeTokenAddress
+        });
+    }
+
+    /// @notice The entrypoint for the CCIP router to call. This function should
+    /// never revert, all errors should be handled internally in this contract.
+    /// @param any2EvmMessage The message to process.
+    /// @dev Extremely important to ensure only router calls this.
+    function ccipReceive(Client.Any2EVMMessage calldata any2EvmMessage) external override onlyRouter {
```

Comment:
is CCIP not able to perform an arbitrary call and it has to call this method?

Reply:
Not from my understanding.